pkg/controllers/vdb/upgrade.go

```diff
@@ -719,3 +720,27 @@ func (i *UpgradeManager) getSubclusterNameFromSts(ctx context.Context, stsName s
 	}
 	return scNameFromLabel, nil
 }
+
+func (i *UpgradeManager) checkAllSubscriptionsActive(ctx context.Context, pfacts *PodFacts) (ctrl.Result, error) {
+	pf, ok := pfacts.findFirstPodSorted(func(v *PodFact) bool {
+		return v.isPrimary && v.upNode
+	})
+	if !ok {
+		i.Log.Info("No pod found to run vsql. Requeueing")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	cmd := []string{"-tAc", "SELECT count(*) FROM node_subscriptions WHERE subscription_state not in ('ACTIVE', 'INACTIVE');"}
```

Comment:
Can the subscriptions be "INACTIVE"? Why not using "subscription_state != 'ACTIVE'"?

Reply:
running a test, yes. subscriptions can be inactive and stay that way. for our use case we don't really care, this check is enough to make sure that sandbox can be run successfully

Reply:
My understanding is: if subscriptions on the nodes of a subcluster are inactive, the subcluster doesn't have coverage to do the sandbox. What do I miss?

Reply:
i'm not sure. all i know is that running rebalance_shards didn't change the subscription_state and sandboxing went fine

Reply:
subscription_state probably is "PENDING" after add_node(before rebalance_shards) so this check will be fine. But the safer way is to verify all subscriptions are "ACTIVE".